pokemongo_bot/cell_workers/sniper.py

```diff
@@ -261,32 +240,24 @@ def initialize(self):
             # Validate mode and sources
             if self.mode not in vars(SniperMode).values():
                 raise ValueError("Unrecognized mode: '{}'".format(self.mode))
-            else:
-                # Selected mode is valid. Validate sources if mode is URL
-                if self.mode == SniperMode.URL:
-                    self._log("Validating sources: {}...".format(", ".join([source.url for source in self.sources])))
-
-                    # Create a copy of the list so we can iterate and remove elements at the same time
-                    for source in list(self.sources):
-                        try:
-                            source.validate()
-                            self._log("Source '{}' is good!".format(source.url))
-                        # TODO: On ValueError, remember source and validate later (pending validation)
-                        except (LookupError, ValueError) as exception:
-                            self._error("Source '{}' contains errors. Details: {}. Removing from sources list...".format(source.url, exception))
-                            self.sources.remove(source)
-
-                    # Notify user if all sources are invalid and cant proceed
-                    if not self.sources:
-                        self._error("There is no source available. Disabling Sniper...")
-                        self.disabled = True
+
+            # Build catchable dictionaries, where key = id and value = priority
+            for key, value in self.config.get('catch', {}).iteritems():
+                self.catch_dictionary[Pokemons.id_for(key)] = value
+            for key, name in self.bot.config.vips.iteritems():
+                self.vip_dictionary[Pokemons.id_for(key)] = 0
 
     def is_snipeable(self, pokemon):
         pokeballs_count = self.inventory.get(Item.ITEM_POKE_BALL.value).count
         greatballs_count = self.inventory.get(Item.ITEM_GREAT_BALL.value).count
         ultraballs_count = self.inventory.get(Item.ITEM_ULTRA_BALL.value).count
         all_balls_count = pokeballs_count + greatballs_count + ultraballs_count
 
+        # Skip if already handled
+        if self._is_cached(pokemon):
+            self._trace('{} was already handled! Skipping...'.format(pokemon.get('pokemon_name')))
+            return False
+
```

Comment:
Yes, but this will result in potential repeated vague "Damn! ...." messages. Better left out of is_snipeable so it just cleanly skips.